### PR TITLE
feat(ui5-multi-input): migrate to Typescript

### PR DESCRIPTION
**Pull request by niyap**
Related to: #4337

### CODE REVIEW

File: packages/main/src/MultiInput.ts [hunk 1]
```diff
@@ -13,81 +18,13 @@ import Input from "./Input.js";
...
-		 * @public
-		 */
-		"token-delete": {
-			detail: {
-				token: { type: HTMLElement },
-			},
-		},
-	},
-};
+type MultiInputTokenDeleteEventDetails = {
```
Comment (nnaydenow) [inline packages/main/src/MultiInput.ts:25]: Here in the type we can say that token has type Token and export that type.

```ts
type MultiInputTokenDeleteEventDetails = {
	token: Token;
}
```

Comment (ilhan007) [reply packages/main/src/MultiInput.ts:25]: MultiInputTokenDeleteEventDetail not Details

File: packages/main/src/MultiInput.ts [hunk 2]
```diff
@@ -114,10 +51,71 @@ const metadata = {
...
 class MultiInput extends Input {
-	static get metadata() {
-		return metadata;
-	}
+	/**
+	 * Determines whether a value help icon will be visualized in the end of the input.
+	 * Pressing the icon will fire <code>value-help-trigger</code> event.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.Input.prototype.showValueHelpIcon
```
Comment (ilhan007) [inline packages/main/src/MultiInput.ts:83]: `sap.ui.webc.main.Input.` should be replaced with `sap.ui.webc.main.MultiInput`.

File: packages/main/src/MultiInput.ts [hunk 3]
```diff
@@ -136,20 +134,21 @@ class MultiInput extends Input {
 
 		// Prevent suggestions' opening.
 		this._skipOpenSuggestions = false;
+		this._valueHelpIconPressed = false;
 	}
 
-	valueHelpPress(event) {
+	valueHelpPress() {
 		this.closePopover();
 		this.fireEvent("value-help-trigger", {});
```
[the remark this answers is not in the document]
Comment (niyap) [reply packages/main/src/MultiInput.ts]: Done

File: packages/main/src/MultiInput.ts [hunk 4]
```diff
@@ -136,20 +134,21 @@ class MultiInput extends Input {
...
 
-	showMorePress(event) {
+	showMorePress() {
 		this.expandedTokenizer = false;
 		this.focus();
 	}
 
-	tokenDelete(event) {
-		const focusedToken = event.detail.ref;
+	tokenDelete(e: CustomEvent) {
```
[the remark this answers is not in the document]
Comment (niyap) [reply packages/main/src/MultiInput.ts]: Done

File: packages/main/src/MultiInput.ts [hunk 5]
```diff
@@ -201,55 +201,56 @@ class MultiInput extends Input {
...
-			const isCut = event.key.toLowerCase() === "x";
+			const isCut = e.key.toLowerCase() === "x";
 			const selectedTokens = tokens.filter(token => token.selected);
 
 			if (isCut) {
-				const cutResult = this.tokenizer._fillClipboard("cut", selectedTokens);
+				const cutResult = this.tokenizer._fillClipboard(ClipboardDataOperation.cut, selectedTokens);
 
 				selectedTokens.forEach(token => {
 					this.fireEvent("token-delete", { token });
```
[the remark this answers is not in the document]
Comment (niyap) [reply packages/main/src/MultiInput.ts]: Done

File: packages/main/src/MultiInput.ts (hunk 2 above)
Comment (niyap) [reply packages/main/src/MultiInput.ts]: Done

File: packages/main/src/MultiInput.ts [hunk 6]
```diff
@@ -13,81 +18,13 @@ import Input from "./Input.js";
 import MultiInputTemplate from "./generated/templates/MultiInputTemplate.lit.js";
 import styles from "./generated/themes/MultiInput.css.js";
 import Token from "./Token.js";
-import Tokenizer from "./Tokenizer.js";
+import Tokenizer, { ClipboardDataOperation, TokenizerTokenDeleteEventDetail } from "./Tokenizer.js";
```
Comment (ilhan007) [inline packages/main/src/MultiInput.ts:21]: TokenizerTokenDeleteEventDetail should be imported with `import type`

Comment (niyap) [reply packages/main/src/MultiInput.ts:21]: Done

**Review by ilhan007 — COMMENTED** [review]
I am fine with the change, @nnaydenow I will let you also take a look

**Review by nnaydenow — APPROVED** [review]

**Review by MapTo0 — APPROVED** [review]

**Pull request merged**